tests/test_galaxy_autoencoder.py

```diff
@@ -2,6 +2,7 @@ def test_galaxy_autoencoder_toy_gaussian(model_setup, devices):
     use_cuda = devices.use_cuda
     overrides = {
         "model": "galaxy_net",
+        "model.residual_delay_n_steps": 50 if use_cuda else 0,
```

Comment:
Could you modify this tests so that in the GPU it ensures that we achieve the higher performance you observed after training? You could either check the `max_residual` or the loss itself (whatever seems more stable).